modules/docs/index.js

```diff
@@ -1,52 +1,46 @@
 const { join } = require('path')
-const getPort = require('get-port')
-const startDocsServer = require('./server')
-const degit = require('degit')
-const logger = require('consola').withScope('modules/docs')
-
-const startServer = async function (path, port) {
-  const server = await startDocsServer(path, port)
-
-  logger.success(`Docs server listening on http://localhost:${port}`)
-  return server
-}
+const DocsServer = require('./server')
+const logger = require('consola').withScope('docs/module')
 
 module.exports = async function (moduleOptions) {
-  const options = Object.assign({}, this.options.docs, moduleOptions)
+  const isDev = this.options.dev
+  const isGenerate = this.options._generate
+  const isStart = this.options._start
+  const isBuild = this.options._build
+  const runServer = isDev || isGenerate || isStart
 
-  let port = options.port
-  if (!port) {
-    port = await getPort()
+  const options = {
+    port: 3001,
+    docsDir: join(isDev ? this.options.srcDir : this.options.buildDir, 'docs'),
+    repo: 'nuxt/docs',
+    watch: isDev,
+    ...this.options.docs,
+    ...moduleOptions
   }
-  this.addPlugin({
-    src: join(__dirname, 'plugin.js'),
-    options: {
-      url: `http://localhost:${port}`
-    }
-  })
 
-  this.nuxt.hook('listen', () => startServer(join(this.options.srcDir, 'docs'), port))
-  // We need to run the server in order to serve the docs when running `nuxt generate`
-  this.nuxt.hook('generate:before', async (generator) => {
-    let server
+  // Start docs server
+  if (runServer) {
+    const docsServer = new DocsServer(options)
+    await docsServer.cloneRepo()
```

Comment:
We only want to clone the repo after the production build and before generating.

In development, I prefer them to clone `nuxt/docs` directly so they can directly push when working on the docs :)

See https://github.com/nuxt/nuxtjs.org#setup

Reply:
Yes. This is handled inside `cloneRepo` method. When target dir exists it will ignore cloning.